### ISSUE TITLE
Shift with arrow keys ins't working with AccessibleNumberSpinner

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/scenery-phet#411]
From https://github.com/phetsims/QA/issues/210, @emily-phet said 

> pressing shift+arrow results in the a left or right tweaker button looking depressed, but nothing happens. Shouldn't this move the slider?

Seems most likely that this would have been introduced from button/PressListener work in https://github.com/phetsims/scenery/issues/831.

**Comment by jessegreenberg** [comment]
This wasn't a problem prior to https://github.com/phetsims/scenery/issues/831 because the a11y timer in press listener updated the pressed property at the end of the interval. Now it is possible that the valueProperty of AccessibleSlider may change as a result of a listener on the increased/decreased emitter, and handleKeyDown in AccessibleSlider doesn't handle that case.

**Comment by jessegreenberg** [comment]
There are two problems - the increasedEmitter and decreasedEmitter need to emit whether or not we increase/decrease, there are cases where these need to emit with a step size of zero. I think for clarity they should be renamed to attemptingIncreaseEmitter (or something). And AccessibleSlider needs to emit these before we get the new value from the valueProperty, since listeners on these emitters could change the value.

**Comment by jessegreenberg** [comment]
`increasedEmitter` and `decreasedEmitter` are only used in NumberControl, so this should be an easy rename.

**Comment by jessegreenberg** [comment]
Fixed with the above commit. @mbarlow12 for https://github.com/phetsims/QA/issues/210, coulombs-law will need to bump its sun SHA to https://github.com/phetsims/sun/commit/640455f4424dc985af3ee9a3f48ab00a27593c0a or use a maintenance branch in sun to include the change. If you have any questions we could do this together soon.

**Comment by jessegreenberg** [comment]
Rename handled above, but that shouldn't be needed for the coulombs-law release. @mbarlow12 assigning to you for https://github.com/phetsims/sun/commit/640455f4424dc985af3ee9a3f48ab00a27593c0a

**Comment by mbarlow12** [comment]
@jessegreenberg I worked with JO to add the proper SHA in the branch's dependencies.json and the built sim is working properly. Is the process to release a new RC version once all the bugs discovered in this round are fixed?

**Comment by jessegreenberg** [comment]
Great, thanks! Yes, that is correct. Once all are fixed we will either ask the QA team to spot check that everything has been fixed in the next RC version or start a new RC test from the beginning if the changes required in the fixes are large enough.

**Comment by jessegreenberg** [comment]
Closing this issue since it has been verified as fixed.

**Comment by mbarlow12** [comment]
@jessegreenberg it looks like you forgot to close—please reopen if I've done so in error!

**Comment by KatieWoe** [comment]
In https://github.com/phetsims/QA/issues/214.  The arrow keys look depressed when using shift + arrow, and don't when just using arrow keys.  The movement is the same in both cases.  Is this what was intended?

**Comment by mbarlow12** [comment]
@KatieWoe good catch. Not totally sure on that.

@jessegreenberg I'm noticing the following lines in NumberControl

```js
// a11y - click the left and right arrow buttons when shift keys are down, must be disposed
    var rightButtonListener = function() { self.shiftKeyDown && rightArrowButton.a11yClick(); };
    var leftButtonListener = function() { self.shiftKeyDown && leftArrowButton.a11yClick(); };
```

any reason why we're only calling a11yClick() with the shift key?

**Comment by mbarlow12** [comment]
Also, should we move this to a new issue related to NumberControl instead of NumberSpinner/Picker?

**Comment by arouinfar** [comment]
Good observation @KatieWoe. 

In general, I don't think the arrow keys should look depressed unless the shift modifier is being used. Tweakers will always change the value by the smallest possible amount, which would be analogous to using the shift modifier. For Coulomb's Law, this seems slightly odd because the arrow keys and shift modifier both increment the slider thumb by one unit. However, I think Coulomb's Law is more of a special case than the norm.

It's not uncommon for sliders to make use of `constrainValue` to snap the thumb in larger increments (Pendulum Lab, for example), which would correspond to the press of an arrow key. Alternatively, in the a11y design, we may decide that the arrow keys should move the thumb in somewhat larger steps (Rutherford Scattering).  In both of these cases, the arrow keys and the shift modifier would be moving the slider thumb by a different amount, so I wouldn't expect to see a depressed tweaker when using the arrow keys.

**Comment by arouinfar** [comment]
@jessegreenberg and/or @mbarlow12 if you agree, please feel free to close.

**Comment by jessegreenberg** [comment]
> any reason why we're only calling a11yClick() with the shift key?

Yes, the design decision was that the arrow buttons should look depressed when the shift modifier was used, but should not look pressed when using normal arrow keys. But it is a bit confusing when the shift step and normal step are the same size.

We could also make it so that if the normal keyboard step is the same as the tweaker delta, the arrow buttons always look pressed. But I think that could also feel inconsistent if a two NumberControls are adjacent and one has this case while the other doesn't.

I agree with @arouinfar  and think this can be closed. Thanks for bringing it up @KatieWoe. @mbarlow12 please reopen if you feel differently.
